osu.Game/Overlays/BeatmapSet/BeatmapSetHeaderContent.cs

```diff
@@ -275,12 +241,53 @@ private void load(OverlayColourProvider colourProvider)
 
                     loading.Hide();
 
-                    title.Text = new RomanisableString(setInfo.NewValue.TitleUnicode, setInfo.NewValue.Title);
-                    artist.Text = new RomanisableString(setInfo.NewValue.ArtistUnicode, setInfo.NewValue.Artist);
+                    var titleText = new RomanisableString(setInfo.NewValue.TitleUnicode, setInfo.NewValue.Title);
+                    var artistText = new RomanisableString(setInfo.NewValue.ArtistUnicode, setInfo.NewValue.Artist);
+
+                    title.Clear();
+                    artist.Clear();
+
+                    title.AddLink(titleText, LinkAction.SearchBeatmapSet, titleText);
+
+                    title.AddArbitraryDrawable(new Container
+                    {
+                        AutoSizeAxes = Axes.Both,
+                        Child = externalLink = new ExternalLinkButton
+                        {
+                            Margin = new MarginPadding { Left = 5 },
+                        }
+                    });
```

Comment:
nit: Can avoid wrapping each piece in a container by using an empty drawable spacer:
```csharp
                    title.AddArbitraryDrawable(Empty().With(d => d.Width = 5));
                    title.AddArbitraryDrawable(externalLink = new ExternalLinkButton());
```

Makes code less verbose at least.